src/Google.Cloud.Functions.Invoker/FunctionsExtensions.cs

```diff
@@ -0,0 +1,175 @@
+        /// <param name="services">The service collection to configure.</param>
+        /// <returns>The original service collection, for method chaining.</returns>
+        public static IServiceCollection AddFunctionTarget<TFunction>(this IServiceCollection services) =>
+            services.AddFunctionTarget(typeof(TFunction));
+
+        /// <summary>
+        /// Configures the given service collection to use the specified target function type, which must
+        /// implement one of the Functions Framework function interfaces. 
+        /// </summary>
+        /// <remarks>
+        /// If this method completes successfully, a binding for <see cref="IHttpFunction"/> will definitely
+        /// have been added to the service collection. Other bindings may also be present, in order to adapt
+        /// the function to <see cref="IHttpFunction"/>.
+        /// </remarks>
+        /// <param name="services">The service collection to configure.</param>
+        /// <param name="type">The target function type.</param>
+        /// <returns>The original service collection, for method chaining.</returns>
+        public static IServiceCollection AddFunctionTarget(this IServiceCollection services, Type type) =>
+            HostingInternals.AddServicesForFunctionTarget(services, type);
+
+        /// <summary>
+        /// Configures the web host builder to use the functions startup classes specified by
+        /// assembly attributes.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="assembly">The assembly to query for attributes specifying startup classes.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionsStartups(this IWebHostBuilder builder, Assembly assembly)
+        {
+            foreach (var startup in HostingInternals.GetStartups(assembly))
+            {
+                builder.AddFunctionsStartup(startup);
+            }
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the web host builder to use the specified functions startup classes. This method can be called
+        /// multiple times, and the startups will all be configured.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="startup">The startup class to use.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionsStartup(this IWebHostBuilder builder, FunctionsStartup startup) =>
+            builder.ConfigureServices((context, services) => HostingInternals.AddStartup(context, services, startup));
+
+        /// <summary>
+        /// Configures the given web host builder to use the Functions Framework.
+        /// This method executes any application configurers before adding handlers for fixed paths to
```

Comment:
```suggestion
        /// This method executes any application Startups before adding handlers to
```

Reply:
Doh, thanks.

Reply:
Still unsure whether these should actually be on IServiceCollection - if we accept that they *are* services, that both configure services *and take part in app configuration* then maybe it's okay to put the extension there?

Reply:
The more I read the comments on the methods, the more it feels like they should be on IServiceCollection. Doing that now, and you can argue against it later if you want :)